### PR TITLE
Backtrace Attributes

### PR DESCRIPTION
**Pull request by sethalves**
- add crash annotation for session-id, domain-id, avatar-url, domain-url, current location, username, launched from steam or not, display plugin name, hmd or not, machine fingerprint

https://highfidelity.fogbugz.com/f/cases/13326/Backtrace-reports-should-be-associated-with-user-data

### CODE REVIEW

**Comment by sethalves** [comment]
to test
- use developer menu to crash interface
- find your crash in backtrace
- make sure the annotations are there and make sense

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9042/app_12970_PR_0fdc0da3bce82da51e7952de4dc61203297ab8e8.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9047/app_12970_PR_69c8a481af86a70806cc1dc5982a9e0c5f187b94.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9055/app_12970_PR_201ae4e78e191061b73c8d79404db0e89533949f.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9065/HighFidelity-Beta-PR12970-001e500cbdd19a7002bd440bcb01252dd520b7c5.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9065/HighFidelity-Beta-Interface-PR12970-001e500cbdd19a7002bd440bcb01252dd520b7c5.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9078/app_12970_PR_2b7f37880cd4617e63c153c6f2881625d25c24bd.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9084/app_12970_PR_9581a6cfe003d98a74187f6cc62d206554eba544.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9084/HighFidelity-Beta-PR12970-9581a6cfe003d98a74187f6cc62d206554eba544.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9084/HighFidelity-Beta-Interface-PR12970-9581a6cfe003d98a74187f6cc62d206554eba544.dmg

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9085/app_12970_PR_18f91cac4129089864a1cfb984060c2d0c5860c8.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9085/HighFidelity-Beta-PR12970-18f91cac4129089864a1cfb984060c2d0c5860c8.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9085/HighFidelity-Beta-Interface-PR12970-18f91cac4129089864a1cfb984060c2d0c5860c8.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9085/HighFidelity-Beta-PR12970-18f91cac4129089864a1cfb984060c2d0c5860c8.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9085/HighFidelity-Beta-Interface-PR12970-18f91cac4129089864a1cfb984060c2d0c5860c8.exe

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -4631,6 +4648,9 @@ void Application::saveSettings() const {
     Menu::getInstance()->saveSettings();
     getMyAvatar()->saveData();
     PluginManager::getInstance()->saveSettings();
+
+    auto addressManager = DependencyManager::get<AddressManager>();
+    setCrashAnnotation("address", addressManager->currentFacingAddress().toString().toStdString());
```
Comment (Atlante45) [inline interface/src/Application.cpp]: This only triggers every 30 sec.
You should put that in `DiscoverabilityManager::updateLocation()`, that's where we send the same thing to metaverse and Steam every 5 sec or so.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9140/app_12970_PR_28122e433dea99f29ac564656c680db427fb7fc2.apk

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9143/app_12970_PR_28122e433dea99f29ac564656c680db427fb7fc2.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9143/HighFidelity-Beta-PR12970-28122e433dea99f29ac564656c680db427fb7fc2.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/9143/HighFidelity-Beta-Interface-PR12970-28122e433dea99f29ac564656c680db427fb7fc2.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9143/HighFidelity-Beta-PR12970-28122e433dea99f29ac564656c680db427fb7fc2.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9143/HighFidelity-Beta-Interface-PR12970-28122e433dea99f29ac564656c680db427fb7fc2.exe

**Comment by sethalves** [comment]
https://highfidelity.fogbugz.com/f/cases/13326/Backtrace-reports-should-be-associated-with-user-data

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/9223/app_12970_PR_5e790d489869fb4bdd24d3538a1b79fc3967e0e5.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9223/HighFidelity-Beta-PR12970-5e790d489869fb4bdd24d3538a1b79fc3967e0e5.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/9223/HighFidelity-Beta-Interface-PR12970-5e790d489869fb4bdd24d3538a1b79fc3967e0e5.exe

**Comment by qlimnavarro** [comment]
Currently Testing

**Comment by qlimnavarro** [comment]
Pass @conklin94122 
The attributes appear and make sense. They seem to appear in the attributes section
capture

**Pull request merged**